File: grib/src/main/java/dev/ucdm/grib/inventory/AbstractMCollection.java
/*
 * Copyright (c) 1998-2021 John Caron and University Corporation for Atmospheric Research/Unidata
 * See LICENSE for license information.
 */
package dev.ucdm.grib.inventory;

import java.util.HashMap;
import java.util.Map;

public abstract class AbstractMCollection implements MCollection {
  protected final String name;
  protected Map<String, Object> auxMap = new HashMap<>();

  public AbstractMCollection(String name) {
    this.name = name;
  }

  @Override
  public String getCollectionName() {
    return name;
  }

  @Override
  public String getIndexFilename(String suffix) {
    return getRoot() + "/" + getCollectionName() + suffix;
  }

  @Override
  public Object getAuxInfo(String key) {
    return auxMap.get(key);
  }

  @Override
  public MCollection setAuxInfo(String key, Object value) {
    auxMap.put(key, value);
    return this;
  }
}

File: grib/src/main/kotlin/dev/ucdm/grib/inventory/AbstractMCollection.kt
package dev.ucdm.grib.inventory

import dev.ucdm.grib.common.GribCollectionIndex

abstract class AbstractMCollection(val name : String) {
    private var auxMap = mutableMapOf<String, Any>()

    fun getCollectionName() = name

    open fun getIndexFilename(): String {
        return "${getRoot()}/$name${GribCollectionIndex.NCX_SUFFIX}"
    }

    abstract fun getRoot(): String

    open fun getAuxInfo(key: String): Any? {
        return auxMap[key]
    }

    open fun setAuxInfo(key: String, value: Any) {
        auxMap[key] = value
    }

    open fun addAuxInfo(info: Map<String, Any>) {
        auxMap.putAll(info)
    }
}